server/bin/utils/run-unittest

```
@@ -0,0 +1,607 @@
+        (cd ${_testroot}; ${_state_reset})
+        if [[ ${?} -gt 0 ]]; then
+            echo "ERROR: unable to run per-test state reset for ${1}" >&2
+            exit 1
+        fi
+    fi
+
+    export PATH=${_orig_PATH}
+    export PYTHONPATH=${_orig_PYTHONPATH}
+    unset _PBENCH_SERVER_CONFIG
+    unset SATCONFIG
+
+    rm -f ${_testopt}/lib/config/state-pbench-server.cfg
+    rm -f ${_testactout}
+    rm -f ${_testlog}
+    rm -f ${_testactlog}
+    rm -f ${_testcurlpayload}
+    rm -f ${_testloggerpayload}
+    rm -rf ${_testdir}
+    if [[ -d ${_testdir} ]]; then
+        echo "ERROR: unable to remove pbench hierarchy" >&2
+        exit 1
+    fi
+    rm -rf ${_testdir_local}
+    if [[ -d ${_testdir_local} ]]; then
+        echo "ERROR: unable to remove pbench-local hierarchy" >&2
+        exit 1
+    fi
+    rm -rf ${_testdir_sat}
+    if [[ -d ${_testdir_sat} ]]; then
+        echo "ERROR: unable to remove pbench-satellite hierarchy" >&2
+        exit 1
+    fi
+    rm -rf ${_testdir_sat_local}
+    if [[ -d ${_testdir_sat_local} ]]; then
+        echo "ERROR: unable to remove pbench-satellite-local hierarchy" >&2
+        exit 1
+    fi
+    rm -rf ${_testtmp}
+    if [[ -d ${_testtmp} ]]; then
+        echo "ERROR: unable to remove tmp hierarchy" >&2
+        exit 1
+    fi
+    rm -rf ${_testhtml}
+    if [[ -d ${_testhtml} ]]; then
+        echo "ERROR: unable to remove var-www-html hierarchy" >&2
+        exit 1
+    fi
+    rm -rf ${_testhtml_sat}
+    if [[ -d ${_testhtml_sat} ]]; then
+        echo "ERROR: unable to remove var-www-html hierarchy" >&2
+        exit 1
+    fi
+    rm -rf ${_testroot}/opt
+    if [[ -d ${_testroot}/opt ]]; then
+        echo "ERROR: unable to remove opt hierarchy" >&2
+        exit 1
+    fi
+}
+
+function _run_test {
+    # What it takes to run one test
+    local testname=${1}
+    local cmd=${2}
+    shift 2
+
+    export _testroot="${_testbase}/${testname}"
+    rm -rf ${_testroot}
+    mkdir -p ${_testroot}
```

How are we feeling today about the `-p`?  (I have no problem with it, but lately we've been removing it elsewhere.)

---

Okay, so because of this prompt, I looked at the other uses of `mkdir -p` and found that we were unnecessarily creating a `common` directory.  I've added a separate commit that addresses that along with removing the remaining uses of `mkdir -p` that hide errors.